jena-core/src/main/java/org/apache/jena/reasoner/rulesys/Rule.java

```diff
@@ -675,7 +673,9 @@ private static int nextSplit(int start, boolean white, String line) {
      * @throws ParserException if there is a problem
      */
     public static List<Rule> parseRules(String source,BuiltinRegistry registry) throws ParserException {
-        return parseRules(new Parser(source,registry));
+        BufferedReader reader = new BufferedReader(new StringReader(source));
+        Parser parser = rulesParserFromReader(reader, registry);
+        return parseRules(parser);
```

Comment:
Could we close the reader before returning the created object?

Reply:
Done.